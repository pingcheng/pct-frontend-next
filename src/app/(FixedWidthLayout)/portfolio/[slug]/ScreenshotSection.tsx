import { Heading } from "@/components/Heading";
import styles from "./style.module.css";
import Image from "next/image";

type ScreenshotSectionProps = {
  images: string[];
};

export function ScreenshotSection({ images }: ScreenshotSectionProps) {
  if (images.length === 0) {
    return (
      <div className="w-full card p-8 px-8 rounded-2xl mt-8 bg-opacity-90">
        <div className="mb-8">
          <Heading text="Screenshots" align="center" />
        </div>
        <div className="text-center text-gray-500 py-8">
          No screenshots available for this project.
        </div>
      </div>
    );
  }

  const SCREENSHOT_COLUMNS = 3;
  const columns: string[][] = [];

  images.forEach((image, index) => {
    const columnIndex = index % SCREENSHOT_COLUMNS;
    if (!columns[columnIndex]) {
      columns[columnIndex] = [];
    }
    columns[columnIndex].push(image);
  });

  return (
    <div className="w-full card p-8 px-8 rounded-2xl mt-8 bg-opacity-90">
      <div className="mb-8">
        <Heading text="Screenshots" align="center" />
      </div>
      <div className={styles.portfolioScreenshots}>
        {columns.map((column, index) => {
          return (
            <div key={index}>
              {column.map((image) => {
                return (
                  <div key={image} className={styles.screenshot}>
                    <Image src={image} alt="Screenshot" fill />
                  </div>
                );
              })}
            </div>
          );
        })}
      </div>
    </div>
  );
}
