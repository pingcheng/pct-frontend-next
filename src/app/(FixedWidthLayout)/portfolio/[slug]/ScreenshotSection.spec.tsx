import { render, screen } from "@testing-library/react";
import { ScreenshotSection } from "./ScreenshotSection";

// Mock the next/image component
jest.mock("next/image", () => ({
  __esModule: true,
  default: ({ src, alt, fill }: { src: string; alt: string; fill?: boolean }) => {
    return <img src={src} alt={alt} data-testid="next-image" />;
  },
}));

describe("test <ScreenshotSection />", () => {
  const mockImages = [
    "/images/screenshot1.png",
    "/images/screenshot2.png",
    "/images/screenshot3.png",
    "/images/screenshot4.png",
  ];

  it("should match snapshot", () => {
    const { container } = render(<ScreenshotSection images={mockImages} />);
    expect(container).toMatchSnapshot();
  });

  it("should render nothing when images array is empty", () => {
    const { container } = render(<ScreenshotSection images={[]} />);
    expect(container.firstChild).toBeNull();
  });

  it("should render the 'Screenshots' heading", () => {
    render(<ScreenshotSection images={mockImages} />);
    expect(screen.getByText("Screenshots")).toBeInTheDocument();
  });

  it("should render all provided images", () => {
    render(<ScreenshotSection images={mockImages} />);
    const images = screen.getAllByTestId("next-image");
    expect(images.length).toBe(mockImages.length);
    
    // Verify all image sources are correct
    mockImages.forEach((src) => {
      const imageWithSrc = images.find((img) => img.getAttribute("src") === src);
      expect(imageWithSrc).toBeInTheDocument();
    });
  });

  it("should distribute images across three columns", () => {
    render(<ScreenshotSection images={mockImages} />);
    
    // The component should create 3 column divs
    const columns = screen.getAllByRole("generic").filter(
      (element) => element.parentElement?.className.includes("portfolioScreenshots")
    );
    
    expect(columns.length).toBe(3);
    
    // First column should have 2 images (for 4 total images)
    expect(columns[0].querySelectorAll('[data-testid="next-image"]').length).toBe(2);
    
    // Second column should have 1 image
    expect(columns[1].querySelectorAll('[data-testid="next-image"]').length).toBe(1);
    
    // Third column should have 1 image
    expect(columns[2].querySelectorAll('[data-testid="next-image"]').length).toBe(1);
  });
});
