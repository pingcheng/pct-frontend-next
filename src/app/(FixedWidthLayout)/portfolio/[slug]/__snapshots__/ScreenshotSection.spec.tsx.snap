// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`test <ScreenshotSection /> should match snapshot 1`] = `
<div>
  <div
    class="w-full card p-8 px-8 rounded-2xl mt-8 bg-opacity-90"
  >
    <div
      class="mb-8"
    >
      <h1
        class="heading1 center"
      >
        Screenshots
      </h1>
    </div>
    <div
      class="portfolioScreenshots"
    >
      <div>
        <div
          class="screenshot"
        >
          <img
            alt="Screenshot"
            data-testid="next-image"
            src="/images/screenshot1.png"
          />
        </div>
        <div
          class="screenshot"
        >
          <img
            alt="Screenshot"
            data-testid="next-image"
            src="/images/screenshot4.png"
          />
        </div>
      </div>
      <div>
        <div
          class="screenshot"
        >
          <img
            alt="Screenshot"
            data-testid="next-image"
            src="/images/screenshot2.png"
          />
        </div>
      </div>
      <div>
        <div
          class="screenshot"
        >
          <img
            alt="Screenshot"
            data-testid="next-image"
            src="/images/screenshot3.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
